### PR TITLE
fix incorrect cleaning of use_entities

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix incorrect cleaning of `use_entities`

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

File: rasa/core/domain.py [hunk 1]
```diff
@@ -656,10 +656,10 @@ def cleaned_domain(self) -> Dict[Text, Any]:
         domain_data = self.as_dict()
         for idx, intent_info in enumerate(domain_data["intents"]):
             for name, intent in intent_info.items():
-                if intent.get("use_entities"):
+                if intent.get("use_entities") is True:
                     intent.pop("use_entities")
                 if not intent.get("ignore_entities"):
-                    intent.pop("ignore_entities")
+                    intent.pop("ignore_entities", None)
```
Comment (tabergma) [inline rasa/core/domain.py]: Why did you add the `None` as default return value, if the return value is ignored?

File: tests/core/test_domain.py [hunk 2]
```diff
@@ -501,3 +500,28 @@ def test_load_on_invalid_domain():
     # Currently just deprecated
     # with pytest.raises(InvalidDomain):
     #     Domain.load("data/test_domains/missing_text_for_templates.yml")
+
+
+def test_clean_domain():
+    domain_path = "data/test_domains/default_unfeaturized_entities.yml"
```
Comment (tabergma) [inline tests/core/test_domain.py]: Should we add an intent without a dict value? E.g.
```
intents:
  - mood
```

File: rasa/core/domain.py (hunk 1 above)
Comment (wochinge) [reply rasa/core/domain.py]: cause `pop` throws an error in case the key is not there

**Comment by wochinge** [comment]
Python 3.5 is really not my friend this week.

File: tests/core/test_domain.py (hunk 2 above)
Comment (wochinge) [reply tests/core/test_domain.py]: done
